### ISSUE TITLE
No oncalls returns if `HUBOT_PAGERDUTY_SCHEDULES` is unset

### ISSUE THREAD

**Issue by RussellRollins** [hubot-scripts/hubot-pager-me#118]
I believe this issue was introduced in: https://github.com/hubot-scripts/hubot-pager-me/pull/117

If the `HUBOT_PAGERDUTY_SCHEDULES` environment variable is unset, `allowed_schedules` is an empty array.

I think the intention of the line:

```coffeescript
if !allowed_schedules or schedule.id in allowed_schedules
```

Is to allow _all_ results if the environment variable is unset, however, the !allowed_schedules check is not truthy for empty arrays:

https://repl.it/repls/LoudLimpOolanguage

So instead, it allows _no_ results.

I suspect a better check would be

```coffeescript
if !allowed_schedules.length or schedule.id in allowed_schedules
```

**Comment by stephenyeargin** [comment]
My fault in the code review. 🐼 We were looking towards the completion of the new filters, but I forgot to guard backwards compatibility.

**Comment by stephenyeargin** [comment]
There's also another issue -- if a schedule doesn't have a person assigned to it, it will come through the list regardless.

**Comment by RussellRollins** [comment]
@stephenyeargin no worries, caught the issue today when a hubot got automatically upgraded. I'm happy to open a PR to fix the first issue, but I'm not sure I fully understand the second.

**Comment by stephenyeargin** [comment]
Released in [v3.1.1](https://github.com/hubot-scripts/hubot-pager-me/releases/tag/v3.1.1)
